### PR TITLE
Skip config setting for append-late-data

### PR DESCRIPTION
**Pull request by cyrusv**
S3 and GCS support this feature, but HDFS does not yet

### CODE REVIEW

File: src/main/java/io/confluent/connect/hdfs/DataWriter.java [hunk 1]
```diff
@@ -331,11 +331,11 @@ public void alterSchema(String database, String tableName, Schema schema) {
         topicPartitionWriters.put(tp, topicPartitionWriter);
       }
     } catch (
-        ClassNotFoundException
-            | IllegalAccessException
-            | InstantiationException
-            | InvocationTargetException
-            | NoSuchMethodException e
+    ClassNotFoundException
```
Comment (kkonstantine) [inline src/main/java/io/confluent/connect/hdfs/DataWriter.java]: noise and incorrect indentation by this change.

File: checkstyle/suppressions.xml [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
     <suppress
             checks="NPathComplexity"
-            files="(TopicPartitionWriter).java"
+            files="(TopicPartitionWriter|WALFile).java"
```
Comment (kkonstantine) [inline checkstyle/suppressions.xml]: Not obvious why this change is required here.

**Comment by cricket007** [comment]
Linking to #256

**Comment by cyrusv** [comment]
@cricket007 Thanks for the link!

I left some notes on the other PR, but it seems like "@asdf2014" isn't responding. To summarize, I think the connector does not need the suggested change, since the expected behavior is to rotate only after the certain amount of time has passed, not any time a record comes that does not fit in the current "window". The connector should handle late records intended for a previous window by opening another writer in the `writers` map.

It does raise a good question, maybe @kkonstantine can shed some light on why S3 connector does rotate if partitions do not match, while HDFS does not have this additional condition? Introduced in `kafka-connect-storage-cloud` commit `60eb3b0fc21cc1431e581c9e9dab1f98ea8`.

**Comment by kkonstantine** [comment]
Don't have memory of the differ you mention @cyrusv at the moment. 
But let's not block on this. 
LGTM
